### PR TITLE
Expose /dev/kvm in the guest conversion pod

### PR DESCRIPTION
**Pull request by mansam**
Uses the KVM device plugin to expose /dev/kvm to the guest conversion pod.

Replaces https://github.com/kubevirt/vm-import-operator/pull/421

Signed-off-by: Sam Lucidi <slucidi@redhat.com>

### CODE REVIEW

**Comment by pkliczewski** [comment]
Couple of vmware tests failed with:

> {type: Succeeded, status: False, reason: DataVolumeCreationFailed, message: Error while importing disk image: . Data volume vm-import-e2e-tests-resource-mapping-wcxf9/target-vm-disk-202-1 creation failed: admission webhook "datavolume-validate.cdi.kubevirt.io" denied the request:  Missing Data volume source}

@mansam please check

File: pkg/guestconversion/guestconversion.go [hunk 1]
```diff
@@ -42,9 +43,22 @@ func MakeGuestConversionJobSpec(vmSpec *v1.VirtualMachine, libvirtConfigMap *cor
 							Image:           virtV2vImage,
 							VolumeMounts:    volumeMounts,
 							ImagePullPolicy: imagePullPolicy,
+							// Request access to /dev/kvm via Kubevirt's Device Manager
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
```
Comment (ghost) [inline pkg/guestconversion/guestconversion.go]: IIUC, the `Requests` is not necessary. In my tests, it `Limits` was enough.

Comment (mansam) [reply pkg/guestconversion/guestconversion.go]: Indeed, according to https://kubevirt.io/api-reference/master/definitions.html#_k8s_io_api_core_v1_resourcerequirements it sets the Requests to be equal to Limits if Requests is not set. Thus, this is redundant and can be removed.

File: pkg/guestconversion/guestconversion.go [hunk 2]
```diff
@@ -42,9 +43,19 @@ func MakeGuestConversionJobSpec(vmSpec *v1.VirtualMachine, libvirtConfigMap *cor
 							Image:           virtV2vImage,
 							VolumeMounts:    volumeMounts,
 							ImagePullPolicy: imagePullPolicy,
+							// Request access to /dev/kvm via Kubevirt's Device Manager
+							Resources: corev1.ResourceRequirements{
```
Comment (masayag) [inline pkg/guestconversion/guestconversion.go:47]: @fdupont-redhat Is there a possibility for this request to fail and how will the conversion behave in that case ?

Comment (ghost) [reply pkg/guestconversion/guestconversion.go:47]: If this request fails, the scheduler will not be able to schedule the pod and the job will sit there until it's possible.
This resource request is also used by Kubevirt to launch virtual machines, so I guess that they explored this.

Comment (ghost) [reply pkg/guestconversion/guestconversion.go:47]: And the NodeSelector makes sure that the pod will be on a node where /dev/kvm is present.

**Review by masayag — APPROVED** [review]

**Review by jakub-dzon — APPROVED** [review]

**Review by machacekondra — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**